pkg/providers/aws/cluster_network_provider_test.go

```diff
@@ -733,10 +736,14 @@ func TestNetworkProvider_CreateNetwork(t *testing.T) {
 					}
 				}),
 				ElasticacheApi: buildMockElasticacheClient(func(elasticacheClient *mockElasticacheClient) {
-					elasticacheClient.cacheSubnetGroup = buildElasticacheSubnetGroup()
 					elasticacheClient.modifyCacheSubnetGroupFn = func(input *elasticache.ModifyCacheSubnetGroupInput) (*elasticache.ModifyCacheSubnetGroupOutput, error) {
 						return &elasticache.ModifyCacheSubnetGroupOutput{}, nil
 					}
+					elasticacheClient.describeCacheSubnetGroupsFn = func(input *elasticache.DescribeCacheSubnetGroupsInput) (*elasticache.DescribeCacheSubnetGroupsOutput, error) {
+						return &elasticache.DescribeCacheSubnetGroupsOutput{
+							CacheSubnetGroups: buildElasticacheSubnetGroup(),
```

Comment:
If we have time, we should remove and refactor `buildElasticacheSubnetGroup` to the new override modifyFn pattern.